contracts/epochs/FlowEpoch.cdc

```diff
@@ -147,6 +147,58 @@ access(all) contract FlowEpoch {
         dkgPubKeys: [String]
     )
 
+    /// The Epoch Recover service event is emitted when a recoverEpoch transaction is submitted to the network. When the network 
+    /// gets into an EFM state the recoverEpoch transaction is used to override the current epoch and recover the network without 
+    /// sporking. This service event is processed by protocol participants to update their local protocol state with the new recover
+    /// epoch data.
```

Comment:
As an amendment to [Jordan's suggestion](https://github.com/onflow/flow-core-contracts/pull/440#discussion_r1699188785):
> Protocol State and the FlowEpoch smart contract may have conflicting views about the current
> valid epoch state (epoch counter and phase). The epoch recovery process resolves this inconsistency
> by injecting a new common shared state to proceed forward from. 

I think it would be important to emphasize that the recovery epoch must be a protocol-compliant successor to protocol's state current EFM epoch. I would suggest to add the following to Jordan's suggestion (maybe after line 158)
```golang
/// Note that the Protocol State will only accept a recovery epoch that is a protocol-compliant successor to
/// the protocol's state current EFM epoch. Specifically, this means that the epoch counter and first view for
/// the recovery epoch must be equal to EFM epoch's counter and last view incremented by one. 
```